### ISSUE TITLE
Migrate to JHipster v8

### ISSUE THREAD

**Issue by mraible** [jhipster/jhipster-kotlin#378]
According to https://github.com/jhipster/generator-jhipster/issues/23449, JHipster v8 is getting close. We should migrate and make sure everything works correctly.

**Comment by asomov** [comment]
@sendilkumarn can you please help to identify the work to be done ?

**Comment by sendilkumarn** [comment]
@asomov I have started working on this branch https://github.com/jhipster/jhipster-kotlin/tree/v8.0.0 

There are a quite a few changes in v8 

- [ ]  Server  (WIP @sendilkumarn)
- [ ]  Spring Cache
- [ ]  Spring Data 
- [ ]  Spring WebSocket 
- [ ]  Generator tests 
- [ ]  Integration Tests
- [ ]  Entity Server 
- [ ]  Entity 
- [ ]  Generator fixes
- [ ]  GitHub actions


Feel free to take any item and work on this branch -  https://github.com/jhipster/jhipster-kotlin/tree/v8.0.0

**Comment by mshima** [comment]
@sendilkumarn do you want some help on this?
generator-jhipster v7 -> v8 should be done before any source code change to simplify migration.

**Comment by mshima** [comment]
Looks like kotlin implementation is identical to generator-jhipster except java files is converted to kt files.
Migration plan:
[ ] update cli to match v8 implementation and make ci workflows compatible.
[ ] improve generator-jhipster templates migration.
[ ] migration to v8 using spring-boot v2 templates.

A follow-up task is to update to spring-boot v3 templates.

**Comment by asomov** [comment]
@mshima is your PR complete? I see a number of different branches in your fork.

**Comment by mshima** [comment]
@asomov v8 migration using v7 templates is complete.
Once merged, there are a few follow-ups.
- Update CI updated cli
- Simplify templates synchronization.
- Merge https://github.com/jhipster/jhipster-kotlin/tree/v8.0.0

**Comment by awikatchikaen** [comment]
Hello, any news on when it will be published ?
Thanks

**Comment by mshima** [comment]
@DanielFran bug bounty claim https://opencollective.com/generator-jhipster/expenses/220546.

**Comment by DanielFran** [comment]
@mshima approved
